inginious/frontend/pages/lti.py

```diff
@@ -52,78 +54,175 @@ class LTIBindPage(INGIniousAuthPage):
     def is_lti_page(self):
         return False
 
-    def fetch_lti_data(self, session_id):
-        # TODO : Flask session interface does not allow to open a specific session
-        # It could be worth putting these information outside of the session dict
-        sess = self.database.sessions.find_one({"_id": session_id})
-        if sess:
-            cookieless_session = self.app.session_interface.serializer.loads(want_bytes(sess['data']))
-        else:
-            return KeyError()
-        return session_id, cookieless_session["lti"]
-
-    def GET_AUTH(self):
-        input_data = flask.request.args
-        if "session_id" not in input_data:
-            return self.template_helper.render("lti_bind.html", success=False, session_id="",
-                                               data=None, error=_("Missing LTI session id"))
+    def _get_lti_session_data(self):
+        if not self.user_manager.session_is_lti():
+            return self.template_helper.render("lti_bind.html", success=False, data=None, error=_("Missing LTI session id"))
```

Comment:
The method returns a tuple below, not here. Besides, I would avoid returning rendered templates directly and try to gather these calls to `render` in one unique method, even if this is not the case for most of the existing code.